### DISCUSSION TITLE
Buildpack for Spring Boot

### DISCUSSION THREAD

**Discussion by AmbitiousHERO** [paketo-buildpacks/feedback#211] [Java Team]
https://github.com/paketo-buildpacks/spring-boot
  Build Configuration:
    $BP_SPRING_CLOUD_BINDINGS_DISABLED   false  whether to contribute Spring Boot cloud bindings support
  Launch Configuration:
    $BPL_SPRING_CLOUD_BINDINGS_DISABLED  false  whether to auto-configure Spring Boot environment properties from bindings
    $BPL_SPRING_CLOUD_BINDINGS_ENABLED   true   Deprecated - whether to auto-configure Spring Boot environment properties from bindings
  Creating slices from layers index
    dependencies (87.9 MB)
    spring-boot-loader (258.6 KB)
    snapshot-dependencies (279.0 KB)
    application (596.7 KB)
  Launch Helper: Reusing cached layer
  Spring Cloud Bindings 1.13.0: Contributing to layer
    Downloading from file:///platform/bindings/dependency-mapping/packages/spring-cloud-bindings-1.13.0.jar
    Verifying checksum
    Copying to /layers/paketo-buildpacks_spring-boot/spring-cloud-bindings
  Web Application Type: Reusing cached layer
  4 application slices
  Image labels:
    org.springframework.boot.version
===> EXPORTING
Reusing layer 'paketo-buildpacks/ca-certificates:helper'
Reusing layer 'paketo-buildpacks/bellsoft-liberica:helper'
Reusing layer 'paketo-buildpacks/bellsoft-liberica:java-security-properties'
Reusing layer 'paketo-buildpacks/bellsoft-liberica:jre'
Reusing layer 'paketo-buildpacks/executable-jar:classpath'
Reusing layer 'paketo-buildpacks/spring-boot:helper'
Reusing layer 'paketo-buildpacks/spring-boot:spring-cloud-bindings'
Reusing layer 'paketo-buildpacks/spring-boot:web-application-type'
Reusing layer 'buildpacksio/lifecycle:launch.sbom'
Reusing 5/5 app layer(s)
Reusing layer 'buildpacksio/lifecycle:launcher'
Reusing layer 'buildpacksio/lifecycle:config'
Reusing layer 'buildpacksio/lifecycle:process-types'
Adding label 'io.buildpacks.lifecycle.metadata'
Adding label 'io.buildpacks.build.metadata'
Adding label 'io.buildpacks.project.metadata'
Adding label 'org.springframework.boot.version'
Setting default process type 'web'
Saving sona/sona-schd-starter:1.230811-SNAPSHOT...
*** Images (742f9753b252):
      sona/sona-schd-starter:1.230811-SNAPSHOT
Reusing cache layer 'paketo-buildpacks/bellsoft-liberica:jdk'
Reusing cache layer 'paketo-buildpacks/syft:syft'
Reusing cache layer 'paketo-buildpacks/maven:application'
Reusing cache layer 'paketo-buildpacks/maven:cache'
Reusing cache layer 'paketo-buildpacks/maven:maven'
Reusing cache layer 'buildpacksio/lifecycle:cache.sbom'

My sona-schd project has introduced a common project. When my common project code changes, the code for sona-schd has not changed. I have rebuilt the sona-schd project, which uses caching and believes that there have been no changes. How can I make sona-schd rebuild ,without using caching。

**Comment by dmikusa** [comment]
If you're using `pack build`, there is a `--clear-cache` flag that will do the job. If you're using Spring Boot build tools, you'd need to change the app image name. That'll result in a fresh build.

Alternatively, you could try to force a change to your app in some way. I'm sorry, but I don't really follow the exact scenario here so I can't give you more specifics.

**Reply by AmbitiousHERO** [reply]
Thanks for your help!
